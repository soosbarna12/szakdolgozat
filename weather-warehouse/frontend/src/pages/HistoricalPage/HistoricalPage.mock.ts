import axios from "axios";
import MockAdapter from "axios-mock-adapter";

// Create a mock adapter instance
const mock = new MockAdapter(axios, { delayResponse: 500 });

// Mock the historical data API
mock.onGet("/today/data").reply(200, {
  coord: { lat: 47.4979, lon: 19.0402 },
  main: {
    temp_max: 303.15,
    temp_min: 293.15,
    humidity: 60,
  },
  name: "Budapest",
  state: "Budapest",
  sys: { country: "HU" },
});

// Mock today's weather data API
mock.onGet("/today/data").reply(200, {
  dt: 1696000000,
  main: {
    temp: 298.15,
    feels_like: 300.15,
    humidity: 65,
    pressure: 1012,
  },
  weather: [{ main: "Clear", description: "clear sky" }],
  wind: { speed: 3.6 },
  name: "Budapest",
  sys: { country: "HU" },
});

// Mock the save location API
mock.onPost("/user/saveLocation/").reply(200, {
  message: "Location saved successfully",
});

// Export the mock for use in tests or development
export default mock;