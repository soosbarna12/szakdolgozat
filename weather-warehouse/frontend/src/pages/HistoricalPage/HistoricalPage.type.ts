import dayjs from "dayjs";

export interface DataMapProps {
  title: string;
  content: string;
  location: string;
  handleLocationChange: (newLocation: string) => void;
}

export interface UseHistoricalDataQueryProps {
  location: string;
  date: dayjs.Dayjs | null;
}

export interface UseHistoricalDataProps {
  data: any;
  date: dayjs.Dayjs | null;
}