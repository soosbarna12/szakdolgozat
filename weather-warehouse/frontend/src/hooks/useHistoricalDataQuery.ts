import { useQuery } from "@tanstack/react-query";
import { UseHistoricalDataQueryProps } from "../pages/HistoricalPage/HistoricalPage.type";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";
import { useEffect } from "react";
import { convertServerHistoricalData } from "../utils/dataConverters";

export function useHistoricalDataQuery(props: Readonly<UseHistoricalDataQueryProps>) {
  const { showAlert } = useAlert();
  const { location, date } = props;
  
  const { data, error, isLoading } = useQuery({
    queryKey: ["historicalWeatherData", location, date],
    queryFn: async () => {
      const response = await axios.post(`/historical/historicalData`, {
        location,
        date
      });

      return convertServerHistoricalData(response.data)?.[0] ?? {};
    },
    enabled: !!(location && date),
  });

  useEffect(() => {
    if (error) {
      showAlert("Error fetching historical weather data", "error");
    }
  }, [error]);

  return { data, error, isLoading };
}
