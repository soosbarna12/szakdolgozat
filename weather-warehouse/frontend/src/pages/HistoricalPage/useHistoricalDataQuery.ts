import { useQuery } from "@tanstack/react-query";
import axios from "axios";
import { useAlert } from "../../utils/AlertContext";
import { UseHistoricalDataQueryProps } from "./HistoricalPage.type";
import { useEffect } from "react";

export function useHistoricalDataQuery(props: Readonly<UseHistoricalDataQueryProps>) {
  const { showAlert } = useAlert();
  const { location, date } = props;


  // Fetch weather data for the selected location
  const { data, error } = useQuery({
    queryKey: ["weather", location],
    queryFn: async () => {
      const response = await axios.get(`/today/data?location=${location}&lang=en`);
      if (response.data?.cod === "404") {
        throw new Error("City not found");
      }
      return response.data;
    },
    enabled: !!location,
  });

  
    // Show alert if there is an error fetching data
    useEffect(() => {
        if (error) {
        showAlert("Error fetching data", "error");
        }
    }, [error, showAlert]);


  return { data };
}