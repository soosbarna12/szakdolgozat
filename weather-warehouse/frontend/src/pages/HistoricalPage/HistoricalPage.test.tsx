import '@testing-library/jest-dom';
import { fireEvent, screen, waitFor } from '@testing-library/react';
import React from 'react';
import { useHistoricalDataQuery } from '../../hooks/useHistoricalDataQuery';
import { useSaveLocationQuery } from '../../hooks/useSaveLocationQuery';
import { renderWithQueryClient } from '../../utils/test/renderWithQueryClient';
import { HistoricalPage } from './HistoricalPage';

jest.mock('../../hooks/useHistoricalDataQuery', () => ({
  useHistoricalDataQuery: jest.fn(),
}));

jest.mock('../../hooks/useSaveLocationQuery', () => ({
  useSaveLocationQuery: jest.fn(),
}));

jest.mock('../../utils/exportCSV', () => ({
  exportCSV: jest.fn(),
}));

// Mock react-leaflet components
jest.mock('react-leaflet', () => ({
  MapContainer: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="dataMapContainer">{children}</div>
  ),
  TileLayer: () => <div data-testid="mockTileLayer">Mock TileLayer</div>,
  useMap: jest.fn(() => ({
    setView: jest.fn(),
  })),
}));

describe('pages/HistoricalPage', () => {
  const mockSetLocation = jest.fn();
  const mockSetHistoricalPageData = jest.fn();
  const mockRefetchSaveLocationQuery = jest.fn();
  const mockExportCSV = jest.requireMock('../../utils/exportCSV').exportCSV;

  beforeEach(() => {
    jest.clearAllMocks();
    (useHistoricalDataQuery as jest.Mock).mockReturnValue({
      data: [
        { date: '2023-01-01', lat: 40.7128, lon: -74.006, temperature: 5 },
      ],
      error: null,
      isLoading: false,
    });
    (useSaveLocationQuery as jest.Mock).mockReturnValue({
      refetch: mockRefetchSaveLocationQuery,
    });
  });

  it('matches HistoricalPage component snapshot', () => {
    const component = renderWithQueryClient(<HistoricalPage />);
    expect(component.container).toMatchSnapshot();
  });

  it('renders error message when there is an error', () => {
    (useHistoricalDataQuery as jest.Mock).mockReturnValue({
      data: null,
      error: true,
      isLoading: false,
    });

    renderWithQueryClient(<HistoricalPage />);

    const errorMessage = screen.getByText('Error fetching historical data.');
    expect(errorMessage).toBeInTheDocument();
  });

  it('renders weather card when data is available', () => {
    renderWithQueryClient(<HistoricalPage />);

    const weatherCard = screen.getByTestId('historicalWeatherCard');
    expect(weatherCard).toBeInTheDocument();
  });

  it('handles Save button click in ActionsMenu', async () => {
    renderWithQueryClient(<HistoricalPage />);
  
    // Ensure the ActionsButton is visible
    const actionsButton = screen.getByTestId('actionsButton');
    expect(actionsButton).toBeInTheDocument();
  
    fireEvent.click(actionsButton);
  
    const saveButton = screen.getByTestId('saveMenuItem');
    fireEvent.click(saveButton);
  
    await waitFor(() => {
      expect(mockRefetchSaveLocationQuery).toHaveBeenCalled();
    });
  });

  it('handles Export button click in ActionsMenu', () => {
    renderWithQueryClient(<HistoricalPage />);

    const actionsButton = screen.getByTestId('actionsButton');
    fireEvent.click(actionsButton);

    const exportButton = screen.getByTestId('exportMenuItem');
    fireEvent.click(exportButton);

    expect(mockExportCSV).toHaveBeenCalledWith([
      { date: '2023-01-01', lat: 40.7128, lon: -74.006, temperature: 5 },
    ]);
  });

  it('handles Reset button click in ActionsMenu', () => {
    renderWithQueryClient(<HistoricalPage />);

    const actionsButton = screen.getByTestId('actionsButton');
    fireEvent.click(actionsButton);

    const resetButton = screen.getByTestId('resetMenuItem');
    fireEvent.click(resetButton);

    expect(mockSetLocation).toHaveBeenCalledWith({ name: '', lat: 0, lon: 0 });
    expect(mockSetHistoricalPageData).toHaveBeenCalledWith([]);
  });

  it('renders DataMap with correct data', () => {
    renderWithQueryClient(<HistoricalPage />);

    const dataMap = screen.getByTestId('dataMap');
    expect(dataMap).toBeInTheDocument();
  });

  it('renders DataTable with correct data', () => {
    renderWithQueryClient(<HistoricalPage />);

    const dataTable = screen.getByTestId('dataTable');
    expect(dataTable).toBeInTheDocument();
  });
});