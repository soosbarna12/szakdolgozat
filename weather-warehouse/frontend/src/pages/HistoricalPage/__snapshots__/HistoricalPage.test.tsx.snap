// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/HistoricalPage handles Reset button click in ActionsMenu 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-94iixj-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: none;"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-t4zkma-MuiToolbar-root"
    >
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1tlcqt-MuiAutocomplete-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1hrwdwy-MuiFormControl-root-MuiTextField-root"
          data-testid="locationSearchBox"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1e3qgpi-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1nowbqt-MuiInputAdornment-root"
            >
              <span
                aria-hidden="true"
                class="notranslate"
              >
                ​
              </span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="SearchOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                />
              </svg>
            </div>
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-qve9l1-MuiInputBase-input-MuiOutlinedInput-input"
              id="«r18»"
              placeholder="Location"
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment css-1uhhrmm-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-2y136s-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-w4cd9x"
              >
                <span
                  aria-hidden="true"
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-qkn3r8-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-jupps9-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-13meb6w-MuiInputBase-input-MuiOutlinedInput-input"
            disabled=""
            id="«r1b»"
            inputmode="text"
            placeholder="Date"
            type="text"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-elo8k2-MuiInputAdornment-root"
          >
            <button
              aria-label="Choose date"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="CalendarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                />
              </svg>
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-w4cd9x"
            >
              <span
                aria-hidden="true"
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiBox-root css-2uchni"
        data-testid="actionsButton"
      >
        <button
          aria-label="Open actions"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-16j2zoz-MuiButtonBase-root-MuiButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          Actions
          <span
            class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-1gulhci-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="KeyboardArrowDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </header>
  <div
    class="MuiBox-root css-uasic2"
  >
    <div
      class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 css-ubb9a5-MuiGrid2-root"
    >
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-8 css-xjklfj-MuiGrid2-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1vaexip-MuiPaper-root"
          data-testid="dataMap"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <div
            data-testid="dataMapContainer"
          >
            <div
              data-testid="mockTileLayer"
            >
              Mock TileLayer
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-4 css-cf65nn-MuiGrid2-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1vaexip-MuiPaper-root"
          data-testid="historicalWeatherCard"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1km0t8c-MuiSkeleton-root"
            data-testid="historicalWeatherCardSkeleton"
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-12 css-qm29e0-MuiGrid2-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1vaexip-MuiPaper-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1km0t8c-MuiSkeleton-root"
            data-testid="dataTableSkeleton"
            style="width: 100%; height: 367px;"
          />
        </div>
      </div>
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-6 css-mzz946-MuiGrid2-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1vaexip-MuiPaper-root"
          data-testid="dataTable"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1km0t8c-MuiSkeleton-root"
            data-testid="dataChartSkeleton"
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-6 css-mzz946-MuiGrid2-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1vaexip-MuiPaper-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1km0t8c-MuiSkeleton-root"
            data-testid="precipitationChartSkeleton"
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-6 css-mzz946-MuiGrid2-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1vaexip-MuiPaper-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1km0t8c-MuiSkeleton-root"
            data-testid="windPressureSkeleton"
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`pages/HistoricalPage matches HistoricalPage component snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-94iixj-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: none;"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-t4zkma-MuiToolbar-root"
    >
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1tlcqt-MuiAutocomplete-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1hrwdwy-MuiFormControl-root-MuiTextField-root"
          data-testid="locationSearchBox"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1e3qgpi-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1nowbqt-MuiInputAdornment-root"
            >
              <span
                aria-hidden="true"
                class="notranslate"
              >
                ​
              </span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="SearchOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                />
              </svg>
            </div>
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-qve9l1-MuiInputBase-input-MuiOutlinedInput-input"
              id="«r0»"
              placeholder="Location"
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment css-1uhhrmm-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-2y136s-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-w4cd9x"
              >
                <span
                  aria-hidden="true"
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-qkn3r8-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-jupps9-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-13meb6w-MuiInputBase-input-MuiOutlinedInput-input"
            disabled=""
            id="«r3»"
            inputmode="text"
            placeholder="Date"
            type="text"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-elo8k2-MuiInputAdornment-root"
          >
            <button
              aria-label="Choose date"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="CalendarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                />
              </svg>
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-w4cd9x"
            >
              <span
                aria-hidden="true"
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiBox-root css-2uchni"
        data-testid="actionsButton"
      >
        <button
          aria-label="Open actions"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-16j2zoz-MuiButtonBase-root-MuiButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          Actions
          <span
            class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-1gulhci-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="KeyboardArrowDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </header>
  <div
    class="MuiBox-root css-uasic2"
  >
    <div
      class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 css-ubb9a5-MuiGrid2-root"
    >
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-8 css-xjklfj-MuiGrid2-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1vaexip-MuiPaper-root"
          data-testid="dataMap"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <div
            data-testid="dataMapContainer"
          >
            <div
              data-testid="mockTileLayer"
            >
              Mock TileLayer
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-4 css-cf65nn-MuiGrid2-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1vaexip-MuiPaper-root"
          data-testid="historicalWeatherCard"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1km0t8c-MuiSkeleton-root"
            data-testid="historicalWeatherCardSkeleton"
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-12 css-qm29e0-MuiGrid2-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1vaexip-MuiPaper-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1km0t8c-MuiSkeleton-root"
            data-testid="dataTableSkeleton"
            style="width: 100%; height: 367px;"
          />
        </div>
      </div>
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-6 css-mzz946-MuiGrid2-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1vaexip-MuiPaper-root"
          data-testid="dataTable"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1km0t8c-MuiSkeleton-root"
            data-testid="dataChartSkeleton"
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-6 css-mzz946-MuiGrid2-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1vaexip-MuiPaper-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1km0t8c-MuiSkeleton-root"
            data-testid="precipitationChartSkeleton"
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-md-6 css-mzz946-MuiGrid2-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1vaexip-MuiPaper-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1km0t8c-MuiSkeleton-root"
            data-testid="windPressureSkeleton"
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
